### ISSUE TITLE
Sub-optimal default attribution

### ISSUE THREAD

**Issue by jorisvandenbossche** [darribas/contextily#51]
https://github.com/darribas/contextily/pull/43 added a default attribution text (original discussion about it in https://github.com/darribas/contextily/issues/25).

However, I am opening an issue to argue that this default is not really optimal IMO:

- For many plots, the text is too big and "falls off" the image.

  Eg, just using all defaults and plotting the NYC boroughs gives me: 

    ```
    import contextily
    import geopandas

    df = geopandas.read_file(geopandas.datasets.get_path('nybb'))
    df = df.to_crs(epsg=3857)

    ax = df.plot(ax=ax)
    contextily.add_basemap(ax)
    ```

  gives

  ![test](https://user-images.githubusercontent.com/1020496/50801287-520df280-12e4-11e9-918a-a67fb06c09db.png)

  (which doesn't give a good first impression I think)

* It hardcodes the text, making this only applicable for the default tile provider. 
  Which means that once you specify another provider / map style, you also *need* to specify the attribution text to make your figure not incorrect.

Both are probably fixable (for the first, we might need to somehow detect what the text size can be given the image size, not sure if that is possible with matplotlib. For the second we might need more complex provider definition with additional attributes). 
But personally, until those are fixed, I would remove the default addition of the attribution text (in the idea of better no attribution than a sub-optimal one). To be clear, in that proposal, we of course would keep the generic functionality added in the other PR, so you can still manually add an attribution with `add_attribution` or specifying the text.

Thought on disabling the attribution text by default for now?

**Comment by knaaptime** [comment]
when we get #10 figured out, it should be able to pull the appropriate attribution to address the second bullet

**Comment by darribas** [comment]
Thanks for reporting! I agree, this is annoying and suboptimal. I lean towards your suggestion, if we keep this in mind as we push forward in #10, we could solve this one too. I think ideally, this probably look like revamping the `sources` module to make them all look the same, whether we hardcoded them or not. @ljwolf had a suggestion at some point to create a method to pull the source, maybe we can explore that? I'd personally want to make it in some way that the user could do `ctx.sources. + Tab` and get a list of everything available (mostly because I always forget the specific name of each tile). If we do get a method, it could also return the appropriate license blurb.

**Comment by jorisvandenbossche** [comment]
Closed by #91, #94, #96
